reporter.go

```diff
@@ -207,6 +210,15 @@ func (r *remoteReporter) Report(span *Span) {
 	// comes, while reporter was already closed
 	case <-r.closed:
 		r.reporterOptions.logger.Infof("Span not reported since Reporter is already closed: %+v", span)
+		return
+	default:
+	}
+
+	// Increment a counter for the requests currently in-flight
+	r.reportsInflight.Add(1)
+	defer r.reportsInflight.Done()
```

Comment:
How about this scenario (GR=go routine):
  * GR1 enters Report(), passes the first select, then gets suspended just before `r.reportsInflight.Add(1)`
  * GR2 enters Close(). Nothing is in-flight, it closes `r.queue` and blocks on `r.queueDrained.Wait()`
  * GR1 wakes up, increments in-flight, the tries to write to closed `r.queue` ==> panic

Reply:
Correct, technically this might happen, though highly improbably, since those operations are CPU bound and unlikely to be switched in the middle of execution pipeline. This however doesn't eliminate RC completely.

We can't increment in-flights counter as soon as we entered the method, b/c WG will panic coming from 0 to 1 in case one of the GRs is already waiting on it.

Alternatively we could've made double checking for closure (to make it properly) but that makes code clogged.